File: Python/191.Number of 1 Bits.py
#求32位整数中，按位算，1的个数有多少
class Solution(object):
    def hammingWeight(self, n):
        hanmmingweight = 0
        while n>0:
            if n&1==1:
                hanmmingweight += 1
            n>>=1
        return hanmmingweight

s = Solution()
print(s.hammingWeight(111))

File: C++/Bit Manipulation/191.Number of 1 Bits.cpp
/*
返回32位无符号数中二进制位为1的个数
*/
/*
思路：n&n-1可以把n的最后一个是1的二进制位设置为0。
*/

#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
	int hammingWeight(uint32_t n) {
        int count = 0;
        while(n)
        {
        	n = n & (n-1);
        	++count;
        }
        return count;
    }
};

int main()
{
	Solution s;	
	cout << s.hammingWeight(43261596) <<endl;
}

